core/src/apps/bitcoin/sign_tx/layout.py

```diff
@@ -15,15 +15,32 @@
 if False:
     from typing import Iterator
     from trezor import wire
+    from trezor.messages.SignTx import EnumTypeAmountUnit
     from trezor.messages.TxOutput import TxOutput
 
     from apps.common.coininfo import CoinInfo
 
 _LOCKTIME_TIMESTAMP_MIN_VALUE = const(500_000_000)
 
 
-def format_coin_amount(amount: int, coin: CoinInfo) -> str:
-    return "%s %s" % (format_amount(amount, coin.decimals), coin.coin_shortcut)
+def format_coin_amount(
+    amount: int, coin: CoinInfo, amount_unit: EnumTypeAmountUnit
+) -> str:
+    decimals, shortcut = coin.decimals, coin.coin_shortcut
+    if amount_unit == AmountUnit.SATOSHI:
+        amount *= 1_0000_0000
+        decimals -= 8
```

Comment:
this seems wrong for coins that don't have 8 decimals?

i'd expect:
```
amount *= 10 ** decimals
decimals = 0
```

Reply:
also the legacy code defends against `decimals < 8` etc, that would probably be a good idea here too

Reply:
Fix in 54085406bbe73202d1c656f26d155de2f80f8bf7